core/src/main/java/org/ehcache/events/CacheEventDispatcherImpl.java

```diff
@@ -48,20 +51,25 @@
  * <p>
  * Conversely, sending events to this service inside lock scope, when there are no registered listeners interested in 
  * ordered event delivery is harmless, i.e. event delivery to unordered listeners will still occur.
- * 
- * @author vfunshte
  */
 public class CacheEventDispatcherImpl<K, V> implements CacheEventDispatcher<K, V> {
 
   private static final Logger LOGGER = LoggerFactory.getLogger(CacheEventDispatcherImpl.class);
-  private final StoreListener<K, V> storeListener = new StoreListener<K, V>();
-  private final Store<K, V> store;
-
-  public CacheEventDispatcherImpl(ExecutorService orderedDelivery, ExecutorService unorderedDelivery, Store<K, V> store) {
-    this.orderedDelivery = orderedDelivery;
-    this.unorderedDelivery = unorderedDelivery;
-    this.store = store;
-    storeListener.setEventNotificationService(this);
+  private final ExecutorService unOrderedExectuor;
+  private final ExecutorService orderedExecutor;
+  private final AtomicInteger listenersCount = new AtomicInteger(0);
+  private final AtomicInteger orderedListenerCount = new AtomicInteger(0);
```

Comment:
These are now only accessed under lock, and so can now be replaced with regular ints